tools/integration-test/src/tests/query_packet.rs

```diff
@@ -0,0 +1,87 @@
+use ibc::core::ics04_channel::channel::IdentifiedChannelEnd;
+use ibc::Height;
+use ibc_relayer::chain::counterparty::pending_packet_summary;
+
+use ibc_test_framework::prelude::*;
+use ibc_test_framework::util::random::random_u64_range;
+
+#[test]
+fn test_query_packet_pending() -> Result<(), Error> {
+    run_binary_channel_test(&QueryPacketPendingTest)
+}
+
+pub struct QueryPacketPendingTest;
+
+impl TestOverrides for QueryPacketPendingTest {
+    fn modify_relayer_config(&self, config: &mut Config) {
+        // Disabling clear_on_start should make the relayer not
+        // relay any packet it missed before starting.
+        config.mode.packets.clear_on_start = false;
+        config.mode.packets.clear_interval = 0;
+    }
+
+    fn should_spawn_supervisor(&self) -> bool {
+        false
+    }
+
+    // Unordered channel: will permit gaps in the sequence of relayed packets
+    fn channel_order(&self) -> Order {
+        Order::Unordered
+    }
+}
+
+impl BinaryChannelTest for QueryPacketPendingTest {
+    fn run<ChainA: ChainHandle, ChainB: ChainHandle>(
+        &self,
+        _config: &TestConfig,
+        relayer: RelayerDriver,
+        chains: ConnectedChains<ChainA, ChainB>,
+        channel: ConnectedChannel<ChainA, ChainB>,
+    ) -> Result<(), Error> {
+        let denom_a = chains.node_a.denom();
+
+        let wallet_a = chains.node_a.wallets().user1().cloned();
+        let wallet_b = chains.node_b.wallets().user1().cloned();
+
+        let amount1 = random_u64_range(1000, 5000);
+
+        info!(
+            "Performing IBC transfer with amount {}, which should *not* be relayed",
+            amount1
+        );
+
+        chains.node_a.chain_driver().transfer_token(
+            &channel.port_a.as_ref(),
+            &channel.channel_id_a.as_ref(),
+            &wallet_a.address(),
+            &wallet_b.address(),
+            amount1,
+            &denom_a,
+        )?;
+
+        sleep(Duration::from_secs(1));
+
+        let channel_end = chains.handle_a().query_channel(
+            channel.port_a.as_ref().value(),
+            channel.channel_id_a.as_ref().value(),
+            Height::zero(),
+        )?;
+        let channel_end = IdentifiedChannelEnd::new(
+            channel.port_a.clone().into_value(),
+            channel.channel_id_a.clone().into_value(),
+            channel_end,
+        );
```

Comment:
@soareschen Should there be an easier way to instantiate an `IdentifiedChannelEnd`, maybe as a method of `ConnectedChannel`?

Reply:
Yes, I think we can probably add a method in `ibc_test_framework::relayer::channel::TaggedChannelEndExt`. 

```rust
trait TaggedChannelEndExt<ChainA, ChainB> {
    fn as_identified_channel_end(&self) -> DualTagged<ChainA, ChainB, IdentifiedChannelEnd>;
}
```

Otherwise there is a tagged version of `query_channel` at `ibc_test_framework::relayer::channel::query_channel_end`. We could alternatively add a new function like `query_identified_channel_end` there.

```rust
fn query_identified_channel_end<ChainA: ChainHandle, ChainB>(
    handle: &ChainA,
    channel_id: &TaggedChannelIdRef<ChainA, ChainB>,
    port_id: &TaggedPortIdRef<ChainA, ChainB>,
) -> Result<DualTagged<ChainA, ChainB, IdentifiedChannelEnd>, Error>
```

The tagged version of query channel helps us ensure that we do not mix up the channel identifiers some where along the way.